components/filter/filter.js

```diff
@@ -861,7 +862,7 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 		switch (dimension.type) {
 			case 'd2l-filter-dimension-set':
 				dimension.values.forEach(value => {
-					if (dimension.searchValue === '') {
+					if (dimension.searchValue === '' || dimension.displayAllKeys) {
```

Comment:
We'll have to explicitly check `dimension.searchType === 'automatic' && dimension.searchValue === ''` since we only want to change the manual search type.